### PR TITLE
BatchNormalization to only allocate dummy mean and var in cuDNN path

### PR DESCRIPTION
**Pull request by hvy**
`F.batch_normalization` always allocates and updates dummy mean and var if they aren't given by the caller. Those dummies are only required when calling cuDNN so I changed the logic to condition these allocations/updates.

### CODE REVIEW

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.
